### DISCUSSION TITLE
Cannot override dependencies using `typing.Annotated` without referencing original function

### DISCUSSION THREAD

**Discussion by michaeloliverx** [fastapi/fastapi#9452] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import Depends, FastAPI, status
from fastapi.testclient import TestClient


def get_user() -> str:
    return "Michael"


GetUserDep = Annotated[str, Depends(get_user)]

app = FastAPI()


@app.get("/")
async def root(user: GetUserDep):
    return {"user": user}


def test_without_override():
    app.dependency_overrides = {}
    test_client = TestClient(app)

    response = test_client.get("/")

    assert response.status_code == status.HTTP_200_OK
    assert response.json() == {"user": "Michael"}


def test_with_override_no_annotated():
    app.dependency_overrides = {get_user: lambda: "Sebastián"}
    test_client = TestClient(app)

    response = test_client.get("/")

    assert response.status_code == status.HTTP_200_OK
    assert response.json() == {"user": "Sebastián"}


def test_with_override_with_annotated():
    app.dependency_overrides = {GetUserDep: lambda: "Sebastián"}
    test_client = TestClient(app)

    response = test_client.get("/")

    assert response.status_code == status.HTTP_200_OK
    assert response.json() == {"user": "Sebastián"}
```

### Description

`test_with_override_with_annotated` fails above, I would expect to be able to override dependencies by specifying the `Annotated`ed version as well as the function reference.

If this is not intended to be the case then a note in the docs would be useful.

### Operating System

macOS

### FastAPI Version

0.95.1

### Python Version

3.11.2

**Comment by Kludex** [comment, marked as answer]
Yeah, this is being causing a bit of confusion. It's kind a same as https://github.com/tiangolo/fastapi/discussions/9674.

So, the dependency is actually `get_user`. The `Annotated` is just "creating a new type".
